package/MDAnalysis/lib/_distopia.py

```diff
@@ -28,48 +28,52 @@
 as a selectable backend.
 """
 import warnings
+from packaging.version import Version
+
+MIN_DISTOPIA_VERSION = Version("0.4.0")
 
 # check for distopia
 try:
     import distopia
 except ImportError:
     HAS_DISTOPIA = False
+    distopia_version = Version("0.0.0")
 else:
     HAS_DISTOPIA = True
 
-    # check for compatibility: currently needs to be >=0.2.0,<0.3.0 (issue
-    # #4740) No distopia.__version__ available so we have to do some probing.
-    needed_funcs = ["calc_bonds_no_box_float", "calc_bonds_ortho_float"]
-    has_distopia_020 = all([hasattr(distopia, func) for func in needed_funcs])
-    if not has_distopia_020:
+    # check for compatibility: currently needs to be >=0.3.1,
+    # some versions of `distopia` don't have a version attribute
+    try:
+        distopia_version = Version(distopia.__version__)
+    except AttributeError:
+        warnings.warn("distopia version cannot be determined, assuming 0.0.0")
```

Comment:
Although we don't need to issue the warning, we need to be able to set the fake distopia version.

Is it worthwhile testing this code path with a proper mock? – @IAlibay ?

Reply:
Test is simple enough through monkeypatch. Adding it would be great, but I'm not sure it needs to be a blocker for this PR. At the end of the day we'll pin the conda-forge recipe to distopia>=0.4 and probably should do the same for pypi (is there a pypi release of distopia?).

TLDR; I'd be ok to see this is as a follow-up PR, but I agree we probably should do it at some point.

Reply:
Can we push to seperate issue? Will raise when this is closed.

> (is there a pypi release of distopia?).

No there isn't, very hard to ship required libs.